plugins/org.eclipse.fordiac.ide.export.ui/src/org/eclipse/fordiac/ide/export/properties/TypeExportPropertyPage.java

```diff
@@ -0,0 +1,250 @@
+				.addModifyListener(e -> directoryEditor.getTextControl(directoryEditorContainer).requestLayout());
+	}
+
+	private String getOutputFolder() {
+		if (getProject().getFolder(OUTPUT_FOLDER_NAME).exists()) {
+			return getProject().getFolder(OUTPUT_FOLDER_NAME).getFullPath().toPortableString();
+		}
+
+		return ""; //$NON-NLS-1$
+	}
+
+	@Override
+	protected IPreferenceStore doGetPreferenceStore() {
+		final ProjectScope projectScope = new ProjectScope(getProject());
+		final ScopedPreferenceStore prefStore = new ScopedPreferenceStore(projectScope,
+				PreferenceConstants.EXPORT_PREFERENCES_ID);
+		prefStore.setDefault(PreferenceConstants.OUTPUT_FOLDER, getOutputFolder());
+		return prefStore;
+	}
+
+	@Override
+	public boolean performOk() {
+		checkboxEditor.store();
+		exporterEditor.store();
+
+		if (!directoryEditor.isValid()) {
+			directoryEditor.showErrorMessage();
+			FordiacLogHelper.logError(directoryEditor.getErrorMessage());
+			return false;
+		}
+		directoryEditor.store();
+		return super.performOk();
+	}
+
+	protected IProject getProject() {
+		return Adapters.adapt(getElement(), IProject.class);
+	}
+
+	private static String[][] getExportFilterNames() {
+		return Stream.of(ExportFilterUtil.getExportFilters())
+				.map(e -> new String[] { e.getAttribute("name"), e.getAttribute("id") }) //$NON-NLS-1$ //$NON-NLS-2$
+				.toArray(size -> new String[size][2]);
+	}
+
+	private void refreshEditors() {
+		checkboxEditor.load();
+		directoryEditor.load();
+		exporterEditor.load();
+		enableSettings(checkboxEditor.getBooleanValue());
+	}
+
+	private void enableSettings(final boolean enable) {
+		settingsContainer.setEnabled(enable);
+		directoryEditor.setEnabled(enable, directoryEditorContainer);
+		exporterEditor.setEnabled(enable, exporterEditorContainer);
+	}
+
+	/*
+	 * Class partly copied from @see DirectoryFieldEditor editor to handle relative
+	 * paths
+	 */
+	class OutputDirectoryFieldEditor extends StringButtonFieldEditor {
+
+		public OutputDirectoryFieldEditor(final String name, final String labelText, final Composite parent) {
+			init(name, labelText);
+			setErrorMessage(JFaceResources.getString("DirectoryFieldEditor.errorMessage"));//$NON-NLS-1$
+			setChangeButtonText(JFaceResources.getString("openBrowse"));//$NON-NLS-1$
+			setValidateStrategy(VALIDATE_ON_FOCUS_LOST);
+			createControl(parent);
+		}
+
+		@Override
+		protected String changePressed() {
+			final Optional<IFolder> selectedDirectory = chooseOutputFolder();
+
+			if (selectedDirectory.isPresent() && selectedDirectory.get().exists()) {
+				return selectedDirectory.get().getProjectRelativePath().toString();
+			}
+			// keep old value
+			return null;
+		}
+
+		@Override
+		protected boolean doCheckState() {
+			String fileName = getTextControl().getText();
+			fileName = fileName.trim();
+			if (fileName.isEmpty() && isEmptyStringAllowed()) {
+				return true;
+			}
+			return getProject().getFolder(new Path(fileName).lastSegment()).exists();
```

Comment:
Why do you only use the last segment here? It works for me with:
```suggestion
			return getProject().getFolder(fileName).exists();
```